File: storydoc-backend/src/main/java/io/storydoc/blueprint/BluePrint.java
package io.storydoc.blueprint;

import io.storydoc.blueprint.classification.Classification;
import io.storydoc.blueprint.element.BluePrintElement;
import io.storydoc.blueprint.element.CompositeBluePrintElement;
import io.storydoc.blueprint.layout.Layout;
import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
public class BluePrint extends CompositeBluePrintElement {

    private List<BluePrintElement> elements;

    @Builder()
    public BluePrint(String name, Layout layout, List<BluePrintElement> elements, Classification classifier) {
        super("BluePrint", name, layout, elements, classifier);
        this.elements = elements;
    }
}

File: storydoc-frontend/src/app/api/models/blue-print.ts
/* tslint:disable */
/* eslint-disable */
import { BluePrintElement } from './blue-print-element';
import { Classification } from './classification';
import { Layout } from './layout';
export interface BluePrint extends BluePrintElement {
  classification?: Classification;
  elements?: Array<BluePrintElement>;
  layout?: Layout;
  name?: string;
  subElements?: Array<BluePrintElement>;
  type?: string;
}
